app/actions/local/thread.ts

```diff
@@ -37,6 +41,35 @@ export const switchToThread = async (serverUrl: string, rootId: string) => {
             return {error: 'Theme not found'};
         }
 
+        // Modal right buttons
+        const rightButtons = [];
+
+        const isCRTEnabled = await getIsCRTEnabled(database);
+        if (isCRTEnabled) {
+            const thread = await post.thread.fetch();
```

Comment:
@avinashlng1080 Will this work as we are expecting here, or will it crash? Asking mainly because of your work in the "channel creator not being defined" PR.

Right now it is in a try/catch block, so the crash is not so bad, but still...

Reply:
Moving it to the remote action and we won't stop the user from opening the thread. In case the thread record isn't available, we can't mark it as unread but again as we immediately insert the thread as we are fetching the thread while we are switching to the thread screen.